### DISCUSSION TITLE
Environment Variable Behavior

### DISCUSSION THREAD

**Discussion by brettonidas** [89luca89/distrobox#1173] [Q&A]
Hi,

Where can I find documentation on the behavior of enviroment variables when entering a containter? I have a container with specific variables setup, but when I run distrobox enter, it seems to overwrite many variables with variables from the host. Is there any way to stop that? 

Thanks!

**Comment by brettonidas** [comment]
I decided to take a look at the code, and I found this in `distrobox-enter`:
```
348     for i in $(printenv | grep '=' | grep -Ev ' |"|`|\$' |
349         grep -Ev '^(HOST|HOSTNAME|HOME|PATH|PROFILEREAD|SHELL|XDG_.*_DIRS|^_)'); do
350         # We filter the environment so that we do not have strange variables,
351         # multiline or containing spaces.
352         # We also NEED to ignore the HOME variable, as this is set at create time
353         # and needs to stay that way to use custom home dirs.
354         result_command="${result_command}
355             --env \"${i}\""
356     done
```

Unless I'm mistaken, that will take all env varibles from the host into the container. The problem is that I'm running a lof the application that use varaibles like `PATH_TO_SOMETHING`. I have them defined to their correct values on my host for my host, and in the containter for the container. However when I use distrobox to enter my container, it blows away the values in the container image with the values from the  host.

Further down it appears to take the PATH from the host, map the paths to the paths in the container, and tack those onto PATH in the container. I imagine that could work for things like scripts or anything that doesn't depend on a shared library, but if I try to run some executable that has RPATH in the ELF header, it's never going to work. Adding the host PATH to the container would mostly be a source of weird problems for me I think.

I would be nice if there were some options to opt-out of this behavior. Or maybe a mode where I can opt-in certain variables to be brought into the containter.

**Reply by jneuhaus** [reply]
I believe I have a similar issue. I build an image using the following Containerfile so I can get jdk8 as part of the baseline image.
```
FROM quay.io/toolbx-images/debian-toolbox:12

RUN apt update && apt -y upgrade 

ENV JAVA_HOME=/opt/java/openjdk
ENV PATH="${JAVA_HOME}/bin:${PATH}"

COPY --from=eclipse-temurin:8 $JAVA_HOME $JAVA_HOME

# Use the host install of podman
RUN ln -s /usr/bin/distrobox-host-exec /usr/local/bin/podman
```

Then, I build a `jdk8` container using: `podman build -t jdk8 -f Containerfile` which succeeds, so I then create the distrobox using:
```
distrobox create \
    --image localhost/jdk8 \
    --home /data/home/jos/box/locator \
    --name locator-jdk8
```
Upon `distrobox enter locator-jdk8`, I echo $PATH and the JAVA_HOME path is missing. I even tried adding `--additional-flags "--env PATH=/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" ` to the distrobox create command, but that didn't help.
